src/components/table/table_pagination/table_pagination.tsx

```diff
@@ -6,137 +6,158 @@
-  activePage?: number;
-  hidePerPageOptions?: boolean;
-  itemsPerPage?: number;
-  itemsPerPageOptions?: number[];
+export interface EuiTablePaginationProps
+  extends Omit<EuiPaginationProps, 'onPageClick'> {
+  /**
+   * Option to completely hide the "Rows per page" selector.
+   */
+  showPerPageOptions?: boolean;
+  /**
+   * Current selection for "Rows per page".
+   * Pass `'all'` to display the selected "Show all" option and hide the pagination.
+   */
+  itemsPerPage?: number | 'all';
+  /**
+   * Custom array of options for "Rows per page".
+   * Pass `'all'` as one of the options to create a "Show all" option.
+   */
+  itemsPerPageOptions?: Array<number | 'all'>;
+  /**
+   * Click handler that passes back selected `pageSize` number
+   */
   onChangeItemsPerPage?: ItemsPerPageChangeHandler;
   onChangePage?: PageChangeHandler;
-  pageCount?: number;
   /**
-   * id of the table being controlled
+   * Requires the `id` of the table being controlled
    */
   'aria-controls'?: string;
   'aria-label'?: string;
 }
 
-interface State {
-  isPopoverOpen: boolean;
-}
-
-export class EuiTablePagination extends Component<
-  EuiTablePaginationProps,
-  State
-> {
-  state = {
-    isPopoverOpen: false,
-  };
+export const EuiTablePagination: FunctionComponent<EuiTablePaginationProps> = ({
+  activePage,
+  itemsPerPage = 50,
+  itemsPerPageOptions = [10, 20, 50, 100],
+  showPerPageOptions = true,
+  onChangeItemsPerPage = () => {},
+  onChangePage,
+  pageCount,
+  ...rest
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
 
-  onButtonClick = () => {
-    this.setState({
-      isPopoverOpen: !this.state.isPopoverOpen,
-    });
-  };
+  const togglePopover = useCallback(() => {
+    setIsPopoverOpen((isOpen) => !isOpen);
+  }, []);
 
-  closePopover = () => {
-    this.setState({
-      isPopoverOpen: false,
-    });
-  };
+  const closePopover = useCallback(() => {
+    setIsPopoverOpen(false);
+  }, []);
 
-  render() {
-    const {
-      activePage,
-      itemsPerPage = 50,
-      itemsPerPageOptions = [10, 20, 50, 100],
-      hidePerPageOptions = false,
-      onChangeItemsPerPage = () => {},
-      onChangePage,
-      pageCount,
-      ...rest
-    } = this.props;
-
-    const button = (
-      <EuiButtonEmpty
-        size="xs"
-        color="text"
-        iconType="arrowDown"
-        iconSide="right"
-        data-test-subj="tablePaginationPopoverButton"
-        onClick={this.onButtonClick}
-      >
-        <EuiI18n
-          token="euiTablePagination.rowsPerPage"
-          default="Rows per page"
-        />
-        : {itemsPerPage}
-      </EuiButtonEmpty>
-    );
+  const button = (
+    <EuiButtonEmpty
+      size="xs"
+      color="text"
+      iconType="arrowDown"
+      iconSide="right"
+      data-test-subj="tablePaginationPopoverButton"
+      onClick={togglePopover}
+    >
+      {itemsPerPage === 'all' ? (
+        <EuiI18n token="euiTablePagination.allRows" default="All rows" />
```

Comment:
I got feedback from @gchaps on the language to use for the button text when this option is selected and settled on:
```suggestion
        <EuiI18n token="euiTablePagination.allRows" default="Showing all rows" />
```